src/main/java/de/tum/in/www1/artemis/service/RatingService.java

```diff
@@ -0,0 +1,67 @@
+package de.tum.in.www1.artemis.service;
+
+import java.util.Optional;
+
+import javax.transaction.Transactional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Rating;
+import de.tum.in.www1.artemis.domain.Result;
+import de.tum.in.www1.artemis.repository.RatingRepository;
+import de.tum.in.www1.artemis.repository.ResultRepository;
+
+/**
+ * Service Implementation for managing {@link de.tum.in.www1.artemis.domain.Rating}.
+ */
+@Service
+public class RatingService {
+
+    private final Logger log = LoggerFactory.getLogger(ResultService.class);
+
+    private final RatingRepository ratingRepository;
+
+    private final ResultRepository resultRepository;
+
+    public RatingService(RatingRepository ratingRepository, ResultRepository resultRepository) {
+        this.ratingRepository = ratingRepository;
+        this.resultRepository = resultRepository;
+    }
+
+    /**
+     * Return Rating that refers to Result with id resultId
+     * @param resultId - Id of Result that the rating refers to
+     * @return Rating if it exists else null
+     */
+    public Optional<Rating> findRatingByResultId(Long resultId) {
+        return ratingRepository.findById(resultId);
+    }
+
+    /**
+     * Persist a new Rating
+     * @param rating - Rating that should be persisted
+     * @return persisted Rating
+     */
+    @Transactional
```

Comment:
Please remove transactional here

Reply:
I've tried it in a few ways but I honestly did not figure out how to do it without using transactional. I've come across the book "Spring Boot Persistence Best Practices: Optimize Java Persistence" which recommended the way I've implemented it now.

<img width="1552" alt="image" src="https://user-images.githubusercontent.com/51994246/82840775-0c9b9680-9ed4-11ea-90c9-6028aa2b57e6.png">

Reply:
The book does not matter, we have design and coding guidelines in Artemis. Please do not use transactional except you have a very good reason for it. I don’t see a good reason in the cases of this PR. Make sure to fetch objects (with related objects) directly from the database to avoid additional fetches. There are plenty of examples in the repositories how to do that

Reply:
I think the `@Transactional` is needed in this case, because the Id is not automatically set otherwise.